packages/launch/src/focused-launch/index.tsx

```diff
@@ -57,22 +41,27 @@ const FocusedLaunch: React.FunctionComponent = () => {
 	}, [ hasSelectedDomain, domainSuggestionFromCart, setDomain ] );
 
 	// @TODO: extract to some hook for reusability (Eg: use-products-from-cart)
-	// If there is no selected plan, but there is a plan in cart,
+	// If there is no selected plan and the site has no paid plan, but there is a plan in cart,
 	// set the plan from cart as the selected plan.
 	const planProductIdFromCart = usePlanProductIdFromCart();
 	const { setPlanProductId } = useDispatch( LAUNCH_STORE );
 
 	React.useEffect( () => {
-		if ( ! selectedPlanProductId && planProductIdFromCart ) {
+		if ( ! selectedPlanProductId && planProductIdFromCart && ! hasPaidPlan ) {
 			setPlanProductId( planProductIdFromCart );
 		}
-	}, [ selectedPlanProductId, planProductIdFromCart, setPlanProductId ] );
+	}, [ selectedPlanProductId, planProductIdFromCart, setPlanProductId, hasPaidPlan ] );
+
+	// If there is a purchased plan, remove any selected plan from Launch store
+	const { unsetPlanProductId } = useDispatch( LAUNCH_STORE );
+	React.useEffect( () => {
+		if ( hasPaidPlan ) {
+			unsetPlanProductId();
+		}
+	}, [ hasPaidPlan, unsetPlanProductId ] );
 
 	return (
-		<Router
-			initialEntries={ [ FocusedLaunchRoute.Summary, FocusedLaunchRoute.Success ] }
-			initialIndex={ shouldDisplaySuccessView ? 1 : 0 }
-		>
+		<Router>
```

Comment:
How is this bit of functionality achieved, instead of using `shouldDisplaySuccessView`?

Reply:
We don't need to show a "persistent" Success view anymore since the user is never leaving the editor as part of the paid flow. After checkout, modal is closed and the site is immediately launched.

Reply:
> We don't need to show a "persistent" Success view anymore since the user is never leaving the editor as part of the paid flow. After checkout, modal is closed and the site is immediately launched.

@StefanNieuwenhuis just to double check with you, today in our review, I remember being redirected to a separate page for the checkout (like wordpress.com/....), instead of having a modal open on top of focused launch.

Could it be that, while you were testing it, your page got redirected to the out of iFrame version of the block editor?

Reply:
Are you sure you were on `calypso.localhost...` or live link when that happend? For wp-admin we still redirect to `/checkout` since there is no Checkout modal and no iframe.

Reply:
Yes, we were checking it on `calypso.localhost`. I've recorded the behavior here: https://www.loom.com/share/4c3655823ed8444c9d35e497304832fb